native/src/main/java/org/ballerinax/asb/receiver/ReceiverNetworkThreadFactory.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KINDither express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinax.asb.receiver;
+
+import java.util.concurrent.ThreadFactory;
+
+/**
+ * A {@link ThreadFactory} object that creates new threads on demand for ASB message-receiver network actions.
+ */
+public class ReceiverNetworkThreadFactory implements ThreadFactory {
+    private final String threadGroupName = "receiver-network-thread";
```

Comment:
Wouldn't it be better to have a asb specific name for troubleshooting purposes?
```suggestion
    private final String threadGroupName = "asb-receiver-network-thread";
```

Reply:
Fixed with [25632cc263a76d7219110ee4c4d0dfa24d1cfd76](https://github.com/ballerina-platform/module-ballerinax-azure-service-bus/pull/230/commits/25632cc263a76d7219110ee4c4d0dfa24d1cfd76)